src/app/core/feedback/feedback.service.js

```diff
@@ -56,16 +70,54 @@ const search = async (reqUser, queryParams, query) => {
 	const offset = page * limit;
 
 	// Query for feedback
-	const feedback = await Feedback.textSearch(query, null, limit, offset, sortArr, true, {
-		path: 'creator',
-		select: ['name', 'email']
-	});
+	const feedback = await Feedback.textSearch(
+		query,
+		null,
+		limit,
+		offset,
+		sortArr,
+		true,
+		{
+			path: 'creator',
+			select: ['username', 'organization', 'name', 'email']
+		}
+	);
 
 	return util.getPagingResults(limit, page, feedback.count, feedback.results);
 };
 
+const readFeedback = async (feedbackId, populate = []) => {
+	if (!mongoose.Types.ObjectId.isValid(feedbackId)) {
+		throw { status: 400, type: 'validation', message: 'Invalid feedback ID' };
```

Comment:
@jrassa @brianghig This and the 404 below feel weird. Is there a better way to do this?

Reply:
Oh nevermind, it looks like this is the way to do it! Resolved!